pandora-common/src/assets/appearanceActions.ts

```diff
@@ -366,22 +328,14 @@ export function DoAppearanceAction(
 				return processingContext.invalid();
 			const item = target.getItem(action.item);
 			// To manipulate the color of room devices, player must be an admin
-			if (item?.isType('roomDevice') && !playerRestrictionManager.isCurrentSpaceAdmin()) {
-				processingContext.addProblem({
-					result: 'restrictionError',
-					restriction: {
-						type: 'modifyRoomRestriction',
-						reason: 'notAdmin',
-					},
-				});
+			if (item?.isType('roomDevice')) {
+				processingContext.checkPlayerIsSpaceAdmin();
 			}
 			// Player coloring the item must be able to interact with the item
-			const r = playerRestrictionManager.canUseItem(processingContext, target, action.item, ItemInteractionType.STYLING);
-			if (!r.allowed) {
-				processingContext.addProblem({
-					result: 'restrictionError',
-					restriction: r.restriction,
-				});
+			processingContext.checkCanUseItem(target, action.item, ItemInteractionType.STYLING);
+
+			if (target.type === 'character') {
+				processingContext.addInteraction(target.character, 'changeItemColor');
 			}
```

Comment:
Thought: Maybe worth moving this interaction check into `canUseItem` when the interaction type is `STYLING`? That type is only used for coloring right now (and in theory for modules that mark themselves as styling, which makes sense for some modules like light/dark variant).